tests/integration/integration_test.go

```diff
@@ -695,3 +694,43 @@ func testComponentProviderSchema(t *testing.T, path string, env ...string) {
 		})
 	}
 }
+
+// Test remote component inputs properly handle unknowns.
+// nolint: unused,deadcode
+func testConstructUnknown(t *testing.T, lang string, dependencies ...string) {
+	const testDir = "construct_component_unknown"
+	tests := []struct {
+		componentDir string
+		env          []string
+	}{
+		{
+			componentDir: "testcomponent",
+		},
+		{
+			componentDir: "testcomponent-python",
+			env:          []string{pulumiRuntimeVirtualEnv(t, filepath.Join("..", ".."))},
+		},
+		{
+			componentDir: "testcomponent-go",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.componentDir, func(t *testing.T) {
+			pathEnv := pathEnv(t,
+				filepath.Join("..", "testprovider"),
+				filepath.Join(testDir, test.componentDir))
+			integration.ProgramTest(t, &integration.ProgramTestOptions{
```

Comment:
Cool. Interesting so this was just bowing up before the change, no need for extra asserts?

Reply:
Right. Before the change, in the case of the Go-based provider, the `Apply` inside the provider's component implementation will run because it thinks the value is known, which will panic and fail the test. With the fix, the `Apply` won't run because we properly pass along unknown, so no panic, and the test succeeds. I wasn't really sure if there was a better way to test this from an integration test.